jupyter_sphinx/execute.py

```diff
@@ -546,6 +635,61 @@ def sphinx_abs_dir(env):
         os.path.abspath(env.app.srcdir)
     )
 
+def add_thebelab_library(doctree, env):
+    """Adds the thebelab configuration and library to the doctree"""
+    thebe_config = env.config.jupyter_sphinx_thebelab_config
+    if isinstance(thebe_config, dict):
+        pass
+    elif isinstance(thebe_config, str):
+        if os.path.isabs(thebe_config):
+            filename = thebe_config
+        else:
+            filename = os.path.join(os.path.abspath(env.app.srcdir), thebe_config)
+
+        if not os.path.exists(filename):
+            logger.warning('The supplied thebelab configuration file does not exist')
+            return
+
+        with open(filename, 'r') as config_file:
+            try:
+                thebe_config = json.load(config_file)
+            except ValueError:
+                logger.warning('The supplied thebelab configuration file is not in JSON format.')
+                return
+    else:
+        logger.warning('The supplied thebelab configuration should be either a filename or a dictionary.')
+        return
+
+    # Force config values to make thebelab work correctly
+    thebe_config['predefinedOutput'] = True
+    thebe_config['requestKernel'] = True
+
+    # Specify the thebelab config inline, a separate file is not supported
+    doctree.append(docutils.nodes.raw(
+        text=f'\n<script type="text/x-thebe-config">\n{json.dumps(thebe_config)}\n</script>',
+        format='html'
```

Comment:
Same as above: how does this look for non-HTML writers? I can't remember if there is a specific node type for javascript that is properly skipped for non-HTML writers.

Reply:
Because format is specified, it will not appear in non-HTML writers, that's AFAIK how raw nodes work.